x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockFlashMessageHelpers, mockHttpValues, mockKibanaValues } from '../../../__mocks__';
+import { LogicMounter } from '../../../__mocks__/kea.mock';
+
+import { engines } from '../../__mocks__/engines.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { asRoleMapping } from '../../../shared/role_mapping/__mocks__/roles';
+import { ANY_AUTH_PROVIDER } from '../../../shared/role_mapping/constants';
+
+import { RoleMappingsLogic } from './role_mappings_logic';
+
+describe('RoleMappingsLogic', () => {
+  const { http } = mockHttpValues;
+  const { navigateToUrl } = mockKibanaValues;
+  const { clearFlashMessages, flashAPIErrors, setSuccessMessage } = mockFlashMessageHelpers;
+  const { mount } = new LogicMounter(RoleMappingsLogic);
+  const DEFAULT_VALUES = {
+    attributes: [],
+    availableAuthProviders: [],
+    elasticsearchRoles: [],
+    roleMapping: null,
+    roleMappings: [],
+    roleType: 'owner',
+    attributeValue: '',
+    attributeName: 'username',
+    dataLoading: true,
+    hasAdvancedRoles: false,
+    multipleAuthProvidersConfig: false,
+    availableEngines: [],
+    selectedEngines: new Set(),
+    accessAllEngines: true,
+    selectedAuthProviders: [ANY_AUTH_PROVIDER],
+  };
+
+  const mappingsServerProps = { multipleAuthProvidersConfig: true, roleMappings: [asRoleMapping] };
+  const mappingServerProps = {
+    attributes: ['email', 'metadata', 'username', 'role'],
+    authProviders: [ANY_AUTH_PROVIDER],
+    availableEngines: engines,
+    elasticsearchRoles: [],
+    hasAdvancedRoles: false,
+    multipleAuthProvidersConfig: false,
+    roleMapping: asRoleMapping,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mount();
+  });
+
+  it('has expected default values', () => {
+    expect(RoleMappingsLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('setRoleMappingsData', () => {
+      it('it sets data based on server response from the `mappings` (plural) endpoint', () => {
```

Comment:
Copy nit / duplicate it
```suggestion
      it('sets data based on server response from the `mappings` (plural) endpoint', () => {
```

Reply:
https://github.com/elastic/kibana/pull/94461/commits/59e9d9335d9fecef6e69b1fc13c57e9aae61ff65